backend-elasticsearch/src/main/java/org/hibernate/search/v6poc/backend/elasticsearch/search/predicate/impl/BooleanJunctionPredicateBuilderImpl.java

```diff
@@ -38,22 +47,22 @@
 	private Map<Integer, MinimumShouldMatchConstraint> minimumShouldMatchConstraints;
 
 	@Override
-	public ElasticsearchSearchPredicateCollector getMustCollector() {
+	public Consumer<SearchPredicateContributor<Void, ? super ElasticsearchSearchPredicateCollector>> getMustCollector() {
```

Comment:
You could turn this into `public void must(SearchPredicateContributor<Void, ? super ElasticsearchSearchPredicateCollector>)` or something similar, and move the weirdness of turning this to a consumer to the caller; only `org.hibernate.search.v6poc.search.dsl.predicate.impl.BooleanJunctionPredicateContextImpl.OccurContext` should need it.

It's just a suggestion though, it would make the backend code a tad simpler but it's not absolutely necessary.

Reply:
Good idea.